example/adversary/adversary_generation.ipynb

```diff
@@ -28,290 +27,312 @@
     "import matplotlib.pyplot as plt\n",
     "import matplotlib.cm as cm\n",
     "\n",
-    "from mxnet.test_utils import get_mnist_iterator"
+    "from mxnet import gluon"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Build Network\n",
-    "\n",
-    "note: in this network, we will calculate softmax, gradient in numpy"
+    "Build simple CNN network for solving the MNIST dataset digit recognition task"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 17,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
    "source": [
-    "dev = mx.cpu()\n",
-    "batch_size = 100\n",
-    "train_iter, val_iter = get_mnist_iterator(batch_size=batch_size, input_shape = (1,28,28))"
+    "ctx = mx.gpu() if len(mx.test_utils.list_gpus()) else mx.cpu()\n",
```

Comment:
Can you use something like this instead?

```
# Use GPU if available
try:
    mx.test_utils.list_gpus(); ctx = mx.gpu()
except:
    ctx = mx.cpu()
batch_size = 128
```

Otherwise it throws an uncaught exception on CPU only machines.

Reply:
I don't manage to reproduce that exception on my mac, any specific machine this happens on?

Reply:
This won't work on Windows machines AFAIK, list_gpus() calls `nvidia-smi` which is not available by default.

Reply:
Get an uncaught exception on an AWS EC2 C5. As mentioned by @roywei, `nvidia-smi` is not available. So you get:

```
CalledProcessError: Command '['nvidia-smi', '-L']' returned non-zero exit status 9.
```

Reply:
Fixed the utils, this error should be caught and range(0,0) should be returned.